packages/cli/src/cmds/validator/keys.ts

```diff
@@ -59,31 +57,34 @@ export async function getLocalSecretKeys(
       lockFilepath(keystorePath);
     }
 
-    const secretKeys = await Promise.all(
-      keystorePaths.map(async (keystorePath) => {
-        const keystoreStr = fs.readFileSync(keystorePath, "utf8");
-
-        let keystore;
-        try {
-          keystore = Keystore.parse(keystoreStr);
-        } catch (e) {
-          (e as Error).message = `Error parsing keystore at ${keystorePath}: ${(e as Error).message}`;
-          throw e;
-        }
-
-        return bls.SecretKey.fromBytes(await keystore.decrypt(passphrase));
-      })
-    );
+    const secretKeys: SecretKey[] = [];
+    // Performing decryption of keystore sequentially as concurrently calls to
+    // keystore.decrypt leads to OOM. See https://github.com/ChainSafe/lodestar/issues/4166
+    // Once the underlining issue causing the OOM is identified and fixed, the
+    // parsing can be updated to be done concurrently
+    for (const keystorePath of keystorePaths) {
+      const keystoreStr = fs.readFileSync(keystorePath, "utf8");
+      try {
+        const keystore = Keystore.parse(keystoreStr);
+        secretKeys.push(bls.SecretKey.fromBytes(await keystore.decrypt(passphrase)));
+      } catch (e) {
+        console.log(`Error parsing keystore file: ${path.basename(keystorePath)}: ${(e as Error).message}`);
```

Comment:
Imo this should be done in two steps so we can retain the proper error messages. As it stands, an error in decrypting a keystore will lead to the error message: "Error parsing keystore file: ..."

Reply:
> Imo this should be done in two steps so we can retain the proper error messages. As it stands, an error in decrypting a keystore will lead to the error message: "Error parsing keystore file: ..."

Updated as suggested!